components/core/src/clp/clg/CommandLineArguments.cpp

```diff
@@ -195,14 +188,13 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
             return ParsingResult::InfoCommand;
         }
 
-        // Parse and validate global metadata DB config
-        if (false == global_metadata_db_config_file_path.empty()) {
-            try {
-                m_metadata_db_config.parse_config_file(global_metadata_db_config_file_path);
-            } catch (std::exception& e) {
-                SPDLOG_ERROR("Failed to validate metadata database config - {}", e.what());
-                return ParsingResult::Failure;
-            }
+        // Initialize and validate global metadata DB config
+        try {
+            m_metadata_db_config.read_credentials_from_env();
```

Comment:
nit: 
if I only read this piece of code, it gives me the feeling that m_metadata_db_config will always read credentials, which is not true. I think there are few alternative:
1. rename `read_credentials_from_env` to something like `read_credentials_from_env_if_needed`
2. explictily check the m_metadata_db_config's type and put read_credentials_from_env under a if-clause.

Reply:
now addressed as responded at https://github.com/y-scope/clp/pull/1148#discussion_r2250524882